pythonforandroid/bootstraps/common/build/templates/Service.tmpl.java

```diff
@@ -34,6 +36,44 @@ static public void start(Context ctx, String pythonServiceArgument) {
         ctx.startService(intent);
     }
 
+    @Override
```

Comment:
Something feels off here, I'm not sure :thinking:
We're duplicating and changing logic of code from the `start()` method above, don't we?
So that I understand. This is when the service gets auto-restarted, it goes through the `onStartCommand()` with a null intent, hence we don't have access to the extra parameters. That's why you're preparing the `nativeStart()` parameters from here, currect?
In this case, do we ever need the logic we have in `start()` and `onStartCommand()`? I mean can't we always asume a null intent and have only one logic in one place?
Another thing I'm thinking we could investigate is the `START_REDELIVER_INTENT` flag maybe.

Reply:
There is one more functionality that is delivered by this PR -> service will auto-restart and will not crash when  main acitivty(app) is closed.  `START_REDELIVER_INTENT` does not seem to support this functionality, so I am keeping `START_STICKY` for now. This PR is battle-tested and has been used for quite some time now, so I am hesitant to make major revisions, maybe soneone can look into `REDELIEVER_INTENT`  at some  later point. 

 I believe `intent` does not always have to be null, in case someone is passing parameters from the app.  I managed to clean up the `run()` a bit, but I am cautious to make any 'invasive' changes, since I am not sure I know of all the usecases this code is used for.